WalletWasabi/Services/MemPoolService.cs

```diff
@@ -26,6 +26,32 @@ public MemPoolService()
 
 		private long _cleanupInProcess;
 
+		public void TryPerformMempoolCleanup(IEnumerable<uint256> allMempoolHashes)
+		{
+			if (Interlocked.Read(ref _cleanupInProcess) == 1) return; // If already cleaning, then no need to run it that often.
+			Interlocked.Exchange(ref _cleanupInProcess, 1);
+			// This function is designed to prevent forever growing mempool.
```

Comment:
This is how you use `Interlocked` as a lock.
```csharp
if(Interlocked.CompareExchange(ref _cleanupInProcess, 1, 0) == 0)
{
	try
	{
						
	}
	finally
	{
		Interlocked.Decrement(ref _cleanupInProcess);
	}
}
```

Reply:
`CompareExchange` is the proper way, it make sure that if `_cleanupInProcess` was 0, then it is now 1. And if it was 1, nothing happen and the if is not executed.

Reply:
I opened an issue for refactoring Interlocked usage. The pattern has been heavily used all over the code. https://github.com/zkSNACKs/WalletWasabi/issues/1276